### PR TITLE
Issue 24297 allow user to edit page or master contentlet in inline editing

### LINKED ISSUE #24297: Allow user to edit page or master contentlet in inline editing
### Parent Issue

#23225

### User Story

As a user, I want to be able to inline edit a piece of content in a page but be able to choose if it will edit for all the pages that use that content or only the one I am working on.

### Mock
[Figma Prototype](https://www.figma.com/proto/trbwjLuf5F6KaU1rFucNeg/Portlets-%2F-Edit-Page?page-id=0%3A1&node-id=569%3A4774&viewport=4311%2C-717%2C0.42&scaling=min-zoom&starting-point-node-id=569%3A4774)

image

### Acceptance Criteria

- [ ] Show the dialog only when the content is used in multiples pages
- [ ] Allow the user to edit the contentlet for this page or all pages.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

We are doing the exact same thing as here: https://github.com/dotCMS/core/issues/23225

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Allow user to edit page or master `contentlet` in inline editing

### Checklist
- [x] Tests
- [x] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### <samp>🤖 Generated by Copilot at fda9f65</samp>

This pull request adds the functionality to show a copy content modal when editing a contentlet that is in multiple pages, and to copy the contentlet to the current page if the user chooses to do so. It also refactors and cleans up some code related to the `dot-edit-content` module, the `DotEditContentHtmlService` class, the `DotCopyContentService` class, and the `dotcms-models` module. It renames some code identifiers and files to improve clarity and consistency.

## Related Issue
Fixes #24297

# Explanation of Changes

### <samp>🤖 Generated by Copilot at fda9f65</samp>

*  Refactor the copy content logic and UI when editing a contentlet in the dot-edit-page module ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL3),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL10-R9),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL26-R18),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL45),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL56-L59),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL108-L124),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL143-R112),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL484-R448),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL666-L723),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-4b954646d8604d2441f78220527fdfd566b815ae8f3835aee8ff79cdeb69b6cdR8),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-4b954646d8604d2441f78220527fdfd566b815ae8f3835aee8ff79cdeb69b6cdR37),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-4b954646d8604d2441f78220527fdfd566b815ae8f3835aee8ff79cdeb69b6cdL86-R90),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-7ef705f09f4f1baa6c63b6bb0a0c29a96a6dd623c70f2b61d22b4730fd7b0256R1-R16),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-725485e1afb6ffdb35753b228418b3a32aa157473004d2b4023af1ba083067c9R1-R69),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L1-R6),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R13),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L19-R20),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R28),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L40-R47),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L71-R75),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L91-R106),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L106-R122),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L237-R260),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L288-R301),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L307-R320),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L468-R480),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L601-R612),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L670-R702),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R750-R751),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L791-R831),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L938-R974),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L955-R1021),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L967-R1034),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L978-R1046),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-d6966285e25922ac6f72a6baa09fc002303bb54883996dc7d7359a999884743aR48-R54),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-9f0aeeaa0e3957f868d4254fec7d12409764e831a90b6e6cc1a7d53e47b71eeeL10),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-9f0aeeaa0e3957f868d4254fec7d12409764e831a90b6e6cc1a7d53e47b71eeeL84-R141),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL4-R4),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL28-R30),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL40-R40),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL49-R49),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL59-R59),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ac7962364b8d8d0c1e945ae1b4ebfed3d99a6362d27b212dfccbb7a480235b55L8-R8),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ac7962364b8d8d0c1e945ae1b4ebfed3d99a6362d27b212dfccbb7a480235b55L23-R30),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-bc0d11d3a769f5ce0ffae2f0de34cfa6334646b433bb28abc62bba16909f3c9aL4-R4),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-87daf5c18b636bfca607fceec1d06e89a7ba2a188446e969f3c1cdaa8439461eL1-L2),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-87daf5c18b636bfca607fceec1d06e89a7ba2a188446e969f3c1cdaa8439461eL9),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5dcfe3bf5ea0d63b08b96fcea4b82d648bfc2d84400648b2175afc4153e23b13L1-R1))
  * Create a new service class `DotCopyContentModalService` in the `dot-copy-content-modal` service to open the `DotBinaryOptionSelectorComponent` and return an observable of the user's choice to copy or not copy a contentlet ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-7ef705f09f4f1baa6c63b6bb0a0c29a96a6dd623c70f2b61d22b4730fd7b0256R1-R16),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-725485e1afb6ffdb35753b228418b3a32aa157473004d2b4023af1ba083067c9R1-R69),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-4b954646d8604d2441f78220527fdfd566b815ae8f3835aee8ff79cdeb69b6cdR37),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-4b954646d8604d2441f78220527fdfd566b815ae8f3835aee8ff79cdeb69b6cdL86-R90))
  * Inject the `DotCopyContentModalService` and the `DotCopyContentService` in the `DotEditContentHtmlService` class and use them to handle the copy content logic when editing a contentlet that is in multiple pages ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R13),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L106-R122),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L670-R702),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L791-R831),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184))
  * Add a new interface `DotShowCopyModal` in the `dot-contentlets-events.model` file to represent the data structure of the event to show the copy content modal ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-d6966285e25922ac6f72a6baa09fc002303bb54883996dc7d7359a999884743aR48-R54))
  * Emit and subscribe to the `showCopyModal` event in the `DotEditContentHtmlService` class and the `inline-edit-mode.js.ts` file to handle the case when a contentlet is in multiple pages ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L71-R75),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L791-R831),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-9f0aeeaa0e3957f868d4254fec7d12409764e831a90b6e6cc1a7d53e47b71eeeL84-R141))
  * Refactor the `addContentlet`, `updateContentlet` and `triggerIframeAction` methods in the `DotEditContentHtmlService` class to use the `getDotPageContainer` and `replaceHTMLContentlet` helper methods, which were extracted from the original code ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L237-R260),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L288-R301),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L307-R320),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L670-R702),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L938-R974),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184))
  * Refactor the `initInlineEditing` method in the `DotEditContentHtmlService` class to use the `onEditBlockEditor` method to handle the copy content logic when editing a block editor contentlet ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L601-R612),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184))
  * Add the `askToCopy` and `pagePersonalization` properties to the `DotEditContentHtmlService` class and use them to control whether the user should be asked to copy a contentlet or not, and to pass the current page personalization value to the `dotCopyContentService.copyInPage` method ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L91-R106),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L670-R702),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R750-R751),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L967-R1034),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184))
  * Inject the `DotLoadingIndicatorService` in the `DotEditContentHtmlService` class and use it to show and hide the loading indicator when copying a contentlet ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R28),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184))
  * Wrap the method call inside the `ngZone.run` method in the `handleDocumentClick` method of the `DotEditContentHtmlService` class to ensure that the Angular change detection is triggered when the user clicks on a contentlet action button ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L468-R480))
  * Rename the `DotCopyContent` interface to `DotTreeNode` in the `dotcms-models` module and the `dot-copy-content` service to reflect its purpose of representing the data structure of a contentlet in a page tree, and to avoid confusion with the copy content logic ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-bc0d11d3a769f5ce0ffae2f0de34cfa6334646b433bb28abc62bba16909f3c9aL4-R4),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5dcfe3bf5ea0d63b08b96fcea4b82d648bfc2d84400648b2175afc4153e23b13L1-R1),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL4-R4),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL49-R49),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ac7962364b8d8d0c1e945ae1b4ebfed3d99a6362d27b212dfccbb7a480235b55L8-R8),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L19-R20),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7L955-R1021))
  * Rename the `copyContentInPage` method to `copyInPage` in the `DotCopyContentService` class to reflect its purpose of copying a contentlet in a page tree ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL28-R30),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL40-R40),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL59-R59),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ac7962364b8d8d0c1e945ae1b4ebfed3d99a6362d27b212dfccbb7a480235b55L23-R30),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-803b27038de2abbfb0091318308d39df3f46e46241dc50eeb2271abeb1b45ae7R1088-R1184))
  * Remove the unused imports and properties from the `dot-edit-content.component.ts`, `dot-edit-content.module.ts`, `dot-edit-content-html.service.ts`, `dot-iframe-edit-event.model.ts` and `inline-edit-mode.js.ts` files ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL3),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL10-R9),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL26-R18),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL45),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL56-L59),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL108-L124),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL143-R112),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-5468a1da5bc321f18ebcfb514f59cae83534fac3917ad24ad7e24864aa057eebL666-L723),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-9f0aeeaa0e3957f868d4254fec7d12409764e831a90b6e6cc1a7d53e47b71eeeL10),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-87daf5c18b636bfca607fceec1d06e89a7ba2a188446e969f3c1cdaa8439461eL1-L2),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-87daf5c18b636bfca607fceec1d06e89a7ba2a188446e969f3c1cdaa8439461eL9))
* Add tests for the `DotCopyContentModalService` class in the `dot-copy-content-modal.service.spec.ts` file ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-7ef705f09f4f1baa6c63b6bb0a0c29a96a6dd623c70f2b61d22b4730fd7b0256R1-R16))
* Update tests for the `DotCopyContentService` class in the `dot-copy-content.service.spec.ts` file to reflect the renaming of the interface and the method ([link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL4-R4),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL28-R30),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL40-R40),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL49-R49),[link](https://github.com/dotCMS/core/pull/24773/files?diff=unified&w=0#diff-ba69eecea5fda9f1b10e06b509c1c7aa1ffee33265944c8474d16fdc6aa33bdeL59-R59))

### Screenshots

https://user-images.githubusercontent.com/72418962/235188208-cb5fb6e6-d44d-4c53-a36d-eff0e1f42395.mov

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    it('shoud have be called one time with the correct data', () => {
+        spyOn(dialogService, 'open').and.returnValue({
+            onClose: of('')
+        } as DynamicDialogRef);
+        service.open().subscribe();
+        expect(dialogService.open).toHaveBeenCalledTimes(1);
+
+        expect(dialogService.open).toHaveBeenCalledWith(jasmine.any(Function), {
+            header: 'Edit Content',
+            width: '37rem',
```
Comment (manuelrojas) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.spec.ts:79]: Should we create a const for this? @rjvelazco

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+const DYNAMIC_DIALOG_CONFIG = {
+    header: 'Edit Content',
+    width: '37rem',
+    data: { options: CONTENT_EDIT_OPTIONS_MOCK },
+    contentStyle: { padding: '0px' }
+};
+
+fdescribe('DotCopyContentModalService', () => {
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.spec.ts:40]: Delete focus

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    };
+
+    constructor(
+        private dialogService: DialogService,
+        private dotMessageService: DotMessageService
+    ) {}
+
+    /**
+     *
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.ts:44]: Add desc here.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                }
+            ]
+        });
+        service = TestBed.inject(DotCopyContentModalService);
+        dialogService = TestBed.inject(DialogService);
+    });
+
+    it('should be created', () => {
+        expect(service).toBeTruthy();
+    });
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.spec.ts:61]: Not needed.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 5]
```diff
@@ -935,10 +969,9 @@ export class DotEditContentHtmlService {
                 relocateInfo.contentlet,
                 this.currentPage
             )
-            .subscribe((contentletHtml: string) => {
-                const newContentletEl: HTMLElement = this.generateNewContentlet(contentletHtml);
-                container.replaceChild(newContentletEl, contenletEl);
-            });
+            .subscribe((contentletHtml: string) =>
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts:972]: We need a `take(1)` here.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 6]
```diff
@@ -952,7 +985,40 @@ export class DotEditContentHtmlService {
         return <HTMLElement>div.children[0];
     }
 
-    private getCopyContentData(contentlet: HTMLElement, container: HTMLElement): DotCopyContent {
+    /**
+     *
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts:989]: Needs desc.

**Review by fmontes — CHANGES_REQUESTED** [review]
.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-copy-content-modal/dot-copy-content-modal.service.ts:44]: Done!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts (hunk 5 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts:972]: Done!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts (hunk 6 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts:989]: Done!

**Review by fmontes — APPROVED** [review]

**Pull request merged**